recipe/shelf.py

```diff
@@ -389,6 +389,84 @@ def parse_validated_field(fld, selectable):
     return field
 
 
+def parse_sureberus_validated_condition(cond, selectable):
+    """ Convert a validated condition into a SQLAlchemy boolean expression """
+    if cond is None:
+        return
+
+    if 'and' in cond:
+        conditions = []
+        for c in cond.get('and', []):
+            conditions.append(
+                parse_sureberus_validated_condition(c, selectable)
+            )
+        return and_(*conditions)
+
+    elif 'or' in cond:
+        conditions = []
+        for c in cond.get('or', []):
+            conditions.append(
+                parse_sureberus_validated_condition(c, selectable)
+            )
+        return or_(*conditions)
+
+    elif 'field' in cond:
+        field = parse_sureberus_validated_field(cond.get('field'), selectable)
+        _op = cond.get('_op')
+        _op_value = cond.get('_op_value')
+        return getattr(field, _op)(_op_value)
+
+
+def parse_sureberus_validated_field(fld, selectable):
+    """ Converts a validated field to sqlalchemy. Field references are
+    looked up in selectable """
+    if fld is None:
+        return
+
+    field = find_column(selectable, fld['value'])
+
+    operator_lookup = {
+        'add': lambda fld: getattr(fld, '__add__'),
+        'sub': lambda fld: getattr(fld, '__sub__'),
+        'div': lambda fld: getattr(fld, '__div__'),
+        'mul': lambda fld: getattr(fld, '__mul__'),
+    }
+    for operator in fld.get('operators', []):
+        op = operator['operator']
+        other_field = parse_sureberus_validated_field(
+            operator['field'], selectable
+        )
+        field = operator_lookup[op](field)(other_field)
+
+    # Apply a condition if it exists
+    cond = parse_sureberus_validated_condition(
+        fld.get('condition', None), selectable
+    )
+    if cond is not None:
+        field = case([(cond, field)])
+
+    aggr_fn = fld.get('_aggregation_fn', lambda x: x)
+    return aggr_fn(field)
+
+
+def ingredient_from_sureberus_validated_dict(ingr_dict, selectable):
```

Comment:
It looks like `ingredient_from_validated_dict` is completely dead code now (no uses left in this branch), so I'm wondering why you decided to add a new function with a different name instead of replacing the existing stuff? the same goes for all the code in schemas.py and validators.py that still uses cerberus. if the old stuff isn't used, why keep it? (I may very well have missed something in this branch that continues to use the old stuff, just haven't noticed it yet)

Reply:
Good points. I'm going to follow this up with a complete removal of cerberus (almost ready for PR). Mostly I was keeping the old stuff around so I could do some A/B testing.

Reply:
I see I see, cool